internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1578,6 +1559,60 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 			Sensitive:    true,
 			ValidateFunc: validation.StringLenBetween(8, 123),
 		}
+		resource.Schema["image_cleaner_enabled"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		}
+		resource.Schema["image_cleaner_interval_hours"] = &pluginsdk.Schema{
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			Default:      48,
+			ValidateFunc: validation.IntBetween(24, 2160),
+		}
+		resource.Schema["automatic_channel_upgrade"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(managedclusters.UpgradeChannelPatch),
+				string(managedclusters.UpgradeChannelRapid),
+				string(managedclusters.UpgradeChannelStable),
+				string(managedclusters.UpgradeChannelNodeNegativeimage),
+			}, false),
+		}
+		resource.Schema["node_os_channel_upgrade"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(managedclusters.NodeOSUpgradeChannelNodeImage),
+				string(managedclusters.NodeOSUpgradeChannelNone),
+				string(managedclusters.NodeOSUpgradeChannelSecurityPatch),
+				string(managedclusters.NodeOSUpgradeChannelUnmanaged),
+			}, false),
+		}
+	}
+
+	if features.FourPointOhBeta() {
+		resource.Schema["automatic_upgrade_channel"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(managedclusters.UpgradeChannelPatch),
+				string(managedclusters.UpgradeChannelRapid),
+				string(managedclusters.UpgradeChannelStable),
+				string(managedclusters.UpgradeChannelNodeNegativeimage),
+			}, false),
+		}
+		resource.Schema["node_os_upgrade_channel"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(managedclusters.NodeOSUpgradeChannelNodeImage),
+				string(managedclusters.NodeOSUpgradeChannelNone),
```

Comment:
can we remove `None` here?

```suggestion
```

Reply:
Unfortunately we can't since None is not the default value for this property